templates/alertmanager.yaml

```
@@ -44,3 +49,4 @@ spec:
   secrets:
     - alertmanager-k8s-tls
     - alertmanager-k8s-proxy
+{{ .Affinity }}
```

```suggestion
{{- if .Affinity }}
  affinity:
{{ .Affinity | indent 4 }}
{{- end }}
```

I realized that the `podAntiAffinity` field as being added to the Prometheus and AlertManager CRs but it wasn't being set on the pods, this is because it has to be in an `affinity` field. This suggestion makes it work, another alternative that I didn't try was to marhsall the following yaml in the `PopulateAffinityRules` function

```go
map[string]*corev1.Affinity {
  "affinity": affinity,
}
```